vivarium/core/serialize.py

```diff
@@ -1,21 +1,417 @@
+            if unit is not None:
+                unit_data = [d.to(unit) for d in data]
+        else:
+            # The superclass deserialize() uses
+            # deserialize_from_string().
+            unit_data = super().deserialize(data)
+            if unit is not None:
+                unit_data.to(unit)
+        return unit_data
+
+
+class ProcessSerializer(Serializer):
+    """Serializer for processes.
+
+    Currently only supports serialization (for emitting simulation
+    configs).
+    """
+
+    def can_serialize(self, data: Any) -> bool:
+        return isinstance(data, Process)
+
+    def serialize_to_string(self, data: Process) -> str:
+        """Create a dictionary of process name and parameters."""
+        return str(dict(data.parameters, _name=data.name))
+
+    def can_deserialize(self, data: Any) -> bool:
+        return False
+
+    def deserialize(self, data: Any) -> None:
+        raise NotImplementedError(
+            'Processes cannot be deserialized.')
+
+
+class ComposerSerializer(Serializer):
+    """Serializer for composers.
+
+    Currently only supports serialization (for emitting simulation
+    configs).
+    """
+
+    def can_serialize(self, data: Any) -> bool:
+        return isinstance(data, Composer)
+
+    def serialize_to_string(self, data: Composer) -> str:
+        """Create a dictionary of composer name and parameters."""
+        return str(dict(data.config, _name=str(type(data))))
+
+    def can_deserialize(self, data: Any) -> bool:
+        return False
+
+    def deserialize(self, data: Any) -> None:
+        raise NotImplementedError(
+            'Composers cannot be deserialized.')
+
+
+class FunctionSerializer(Serializer):
+    """Serializer for functions.
+
+    Currently only supports serialization (for emitting simulation
+    configs).
+    """
+    def can_serialize(self, data: Any) -> bool:
+        return callable(data)
+
+    def serialize_to_string(self, data: Callable) -> str:
+        return str(data)
+
+    def can_deserialize(self, data: Any) -> bool:
+        return False
+
+    def deserialize(self, data: Any) -> None:
+        raise NotImplementedError(
+            'Functions cannot be deserialized.')
+
+
+class ObjectIdSerializer(Serializer):
+    """Serializer for BSON ObjectIds.
+
+    Currently only supports serialization.
+    """
+    def can_serialize(self, data: Any) -> bool:
+        return isinstance(data, ObjectId)
+
+    def serialize_to_string(self, data: ObjectId) -> str:
+        return str(data)
+
+    def can_deserialize(self, data: Any) -> bool:
+        return False
+
+    def deserialize(self, data: Any) -> None:
+        raise NotImplementedError(
+            'ObjectIds cannot be deserialized.')
 
 
 def composite_specification(
         composite: Composite,
         initial_state: bool = False,
 ) -> dict:
     """Return the serialized specification for a :term:`composite` instance"""
+    warnings.warn(
```

Comment:
Is this equivalent to serializing a composite? The reason for this method was to provide an entirely data representation of a composite, which could ideally be re-initialized again. I was thinking of moving it to `composer.py`.

Reply:
I would like it back because I know there are a few cases of it out there. Maybe just have it call `serialize_value` and also give a depreciation warning? I don't remember why the `_multi_update` key sneaked in there, it really doesn't make sense but I know it showed up. `_schema` is an unfortunate default key of composites.

Reply:
I removed the deprecation warning to keep `composite_specification` as-is